src/core.js

```diff
@@ -217,7 +217,11 @@ jQuery.extend( {
 		// that can be coerced to finite numbers (gh-2662)
 		var type = jQuery.type( obj );
 		return ( type === "number" || type === "string" ) &&
-			( obj - parseFloat( obj ) + 1 ) >= 0;
+
+			// parseFloat NaNs numeric-cast false positives ("")
+			// ...but misinterprets leading-number strings, particularly hex literals ("0x...")
+			// subtraction forces infinities to NaN
+			!isNaN( obj - parseFloat( obj ) );
```

Comment:
The change is derived from https://github.com/jquery/jquery/issues/2781#issuecomment-165633427 , and works as follows:
* since `obj - parseFloat( obj )` includes both a numeric cast and an explicit parse, any non-numeric input will generate at least one `NaN` operand
* infinite values survive the conversion, but subtracting an infinite value from itself yields `NaN`
* `isNaN` (or `isFinite`) discriminate `NaN`, and therefore also numeric input